### PR TITLE
Add rle page

### PR DESCRIPTION
**Pull request by kangpinghuang**

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/rle_page.h [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        for (int i = 0; i < *count; ++i) {
+            _rle_encoder->Put(new_vals[i], 1);
+        }
+        
+        _count += *count;
+        return Status::OK();
+    }
+
+    Status get_dictionary_page(Slice* dictionary_page) override {
+        return Status::NotSupported("get_dictionary_page not implemented");
```
Comment (gaodayue) [inline be/src/olap/rowset/segment_v2/rle_page.h]: how about making it the default implementation of PageBuilder?

File: be/src/olap/rowset/segment_v2/rle_page.h [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    }
+
+    Slice finish() override {
+        // here should Flush first and then encode the count header
+        // or it will lead to a bug if the header is less than 8 byte and the data is small
+        _rle_encoder->Flush();
+        encode_fixed32_le(&_buf[0], _count);
+        return Slice(_buf.data(), _buf.size());
```
Comment (gaodayue) [inline be/src/olap/rowset/segment_v2/rle_page.h]: _finished = true

File: be/src/olap/rowset/segment_v2/rle_page.h (hunk 1 above)
Comment (kangpinghuang) [reply be/src/olap/rowset/segment_v2/rle_page.h]: OK

File: be/src/olap/rowset/segment_v2/rle_page.h (hunk 2 above)
Comment (kangpinghuang) [reply be/src/olap/rowset/segment_v2/rle_page.h]: ok

File: be/src/olap/rowset/segment_v2/rle_page.h [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    }
+
+    Status add(const uint8_t* vals, size_t* count) override {
+        DCHECK(!_finished);
+        DCHECK_EQ(reinterpret_cast<uintptr_t>(vals) & (alignof(CppType) - 1), 0)
+                << "Pointer passed to Add() must be naturally-aligned";
+
+        const CppType* new_vals = reinterpret_cast<const CppType*>(vals);
+        for (int i = 0; i < *count; ++i) {
+            _rle_encoder->Put(new_vals[i], 1);
```
[the remark this answers is not in the document]
Comment (kangpinghuang) [reply be/src/olap/rowset/segment_v2/rle_page.h]: ok

File: be/src/olap/types.cpp [hunk 4]
```diff
@@ -62,6 +62,7 @@ TypeInfoResolver::TypeInfoResolver() {
     add_mapping<OLAP_FIELD_TYPE_TINYINT>();
     add_mapping<OLAP_FIELD_TYPE_SMALLINT>();
     add_mapping<OLAP_FIELD_TYPE_INT>();
+    add_mapping<OLAP_FIELD_TYPE_UNSIGNED_INT>();
```
[the remark this answers is not in the document]
Comment (kangpinghuang) [reply be/src/olap/types.cpp]: ok

**Comment by gaodayue** [comment]
LGTM

**Review by imay — APPROVED** [review]

**Pull request merged**